### ISSUE TITLE
Verification got cancelled by apparently seeing two verification ready events from bob.

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#12193]
Verification got cancelled by apparently seeing two verification ready events from bob.

```
{
  "type": "m.key.verification.cancel",
  "content": {
    "code": "m.unexpected_message",
    "reason": "Unexpected m.key.verification.ready event in phase 3",
    "m.relates_to": {
      "rel_type": "m.reference",
      "event_id": "$bvH2Cpf564wbWiCRTVo7JnIg6aPanFpydl9_wkq73qM"
    }
  }
}
```

see rageshakes

**Comment by bwindels** [comment]
Is this a case of not having #12116 and allowing double submission?

**Comment by bwindels** [comment]
We need more logging for cases like these: https://github.com/vector-im/riot-web/issues/12194

**Comment by bwindels** [comment]
this should be fixed by #12116
